localstack/services/cloudformation/engine/template_deployer.py

```diff
@@ -55,6 +55,49 @@
     model.cloudformation_type(): model for model in get_all_subclasses(GenericBaseModel)
 }
 
+# ---------------------
+# TYPES
+# ---------------------
+
+# Callable here takes the arguments:
+# - resource_props
+# - stack_name
+# - resources
+# - resource_id
+ResourceProp = str | Callable[[dict, str, dict, str], dict]
+ResourceDefinition = dict[str, ResourceProp]
+
+
+class FuncDetailsValue(TypedDict):
+    # Callable here takes the arguments:
+    # - resource_id
+    # - resources
+    # - resource_type
+    # - func
+    # - stack_name
+    function: str | Callable[[str, list[dict], str, Any, str], Any]
+    """Either an api method to call directly with `parameters` or a callable to directly invoke"""
+    # Callable here takes the arguments:
+    # - resource_props
+    # - stack_name
+    # - resources
+    # - resource_id
+    parameters: Optional[ResourceDefinition | Callable[[dict, str, list[dict], str], dict]]
+    """arguments to the function, or a function that generates the arguments to the function"""
+    # Callable here takes the arguments
+    # - result
+    # - resource_id
+    # - resources
+    # - resource_type
+    result_handler: Optional[Callable[[dict, str, list[dict], str], None]]
+    """Take the result of the operation and patch the state of the resources, yuck..."""
+    types: Optional[dict[str, Callable]]
+    """Possible type conversions"""
+
+
+# Type definition for func_details supplied to invoke_function and configure_resource_via_sdk
+FuncDetails = list[FuncDetailsValue] | FuncDetailsValue | Any
```

Comment:
Shouldn't that be a `dict[str, Union[FuncDetailsValue, Callable]]` ? :thinking: 

str could even be a union of the literals "create", "delete" for now, but I don't think that's necessary

Reply:
Thanks, I've added another type that's the result of `get_deploy_templates`.